resonance/nonlinear_systems.py

```diff
@@ -81,84 +80,39 @@ def _array_rhs_eval_func(self):
-        integrator_method = getattr(self, method_name)
+        integrator = getattr(self, 'integrator', 'RK45')
 
-        return integrator_method(initial_conditions, times)
-
-    def _integrate_with_lsoda(self, initial_conditions, times):
-        """This method should return the integration results in the form of
-        odeint.
-
-        Parameters
-        ==========
-        initial_conditions : ndarray, shape(n,)
-            The initial condition of each state.
-        times : ndarray, shape(m,)
-            The monotonically increasing time values.
-
-        """
-        return sp.integrate.odeint(self._array_rhs_eval_func,
-                                   initial_conditions, times)
-
-    def _integrate_with_rungakutta4(self, initial_conditions, times):
-        """4th-order Runge-Kutta integration.
-
-        Parameters
-        ----------
-        initial_conditions : array_like, shape(n,)
-            Initial values of the state variables.
-        times : array, shape(m,)
-            Array of time values at which to solve.
-
-        Returns
-        -------
-        x : ndarray, shape(m, n)
-            Array containing the values of the state variables at the
-            specified time values in `t`.
-
-        """
-
-        def _rk4(t, dt, x, f, args=None):
-            """4th-order Runge-Kutta integration step."""
-            x = np.asarray(x)
-            if args is None:
-                args = []
-            k1 = np.asarray(f(x, t, *args))
-            k2 = np.asarray(f(x + 0.5*dt*k1, t + 0.5*dt, *args))
-            k3 = np.asarray(f(x + 0.5*dt*k2, t + 0.5*dt, *args))
-            k4 = np.asarray(f(x + dt*k3, t + dt, *args))
-            return x + dt*(k1 + 2*k2 + 2*k3 + k4)/6.0
-
-        x = np.zeros((len(times), len(initial_conditions)))
-        x[0, :] = initial_conditions
-        for i in range(1, len(times)):
-            dt = times[i] - times[i-1]
-            x[i] = _rk4(times[i], dt, x[i-1], self._array_rhs_eval_func)
-        return x
+        return sp.integrate.solve_ivp(self._array_rhs_eval_func,
+                                      (times[0], times[-1]),
+                                      initial_conditions,
+                                      t_eval=times,
+                                      method=integrator,
+                                      vectorized=True,
+                                      **kwargs)
 
     def _generate_state_trajectories(self, times):
         """This method should return arrays for position, velocity, and
         acceleration of the coordinates."""
-        int_res = self._integrate_equations_of_motion(times)
+        int_bunch = self._integrate_equations_of_motion(times)
+        int_res = int_bunch.y
 
-        res = self._array_rhs_eval_func(int_res.T, times)
+        res = self._array_rhs_eval_func(times, int_res)
 
-        return int_res[:, 0], int_res[:, 1], res[1, :]
+        return int_res[0], int_res[1], res[1, :]
```

Comment:
Interesting. I'm not sure it was a good idea to swith x,t in the input and spit out the transpose wrt to odeint. There is so much code in the wild that uses the other pattern.